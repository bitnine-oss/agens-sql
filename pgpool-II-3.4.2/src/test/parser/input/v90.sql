# named parameter
SELECT func(a := 7, b := 12);
# order by aggrigator
SELECT array_agg(a ORDER BY b, c);
# variadic from html/xfunc-sql.html
SELECT mleast(VARIADIC ARRAY[10, -1, 5, 4.4]);
# window function
SELECT avg(c1) OVER (ROWS 1 PRECEDING) FROM r1;
SELECT avg(c1) OVER w1 FROM r1 WINDOW w1 AS (ROWS BETWEEN 1 FOLLOWING AND 2 FOLLOWING);
# offset
SELECT * FROM t1 OFFSET 1 ROWS FETCH FIRST 2 ROWS ONLY;
SELECT * FROM t1 FETCH NEXT 2 ROWS ONLY;
# DO
DO LANGUAGE PLPGSQL $$BEGIN RAISE NOTICE 'hello'; END$$;

\pset server_version 9.0.0
# explain
# old syntax
EXPLAIN ANALYZE VERBOSE SELECT 1;
# new syntax
EXPLAIN (ANALYZE true,VERBOSE true,COSTS true,BUFFERS true,FORMAT XML) SELECT 1;
EXPLAIN (ANALYZE false,VERBOSE false,COSTS false,BUFFERS false,FORMAT JSON) SELECT 1;
# copy
# old syntax
COPY t FROM stdin;
COPY t FROM 'xxx' BINARY;
COPY t FROM 'xxx' CSV HEADER QUOTE AS '#' ESCAPE AS '$' FORCE NOT NULL foo,bar;
COPY t FROM 'xxx' OIDS;
COPY t FROM 'xxx' DELIMITER ',';
COPY t FROM 'xxx' NULL 'x';
# new syntax
COPY t FROM stdin (FORMAT 'text');
COPY t FROM stdin (FORMAT 'csv');
COPY t FROM stdin (FORMAT 'binary');
COPY t FROM 'f' (OIDS false);
COPY t FROM 'f' (OIDS true, DELIMITER '|', NULL '<N>', HEADER false);
COPY t FROM 'f' (HEADER true, QUOTE '"', ESCAPE '@');
COPY t FROM 'f' (FORCE_QUOTE *, FORCE_NOT_NULL(one, two, three));
