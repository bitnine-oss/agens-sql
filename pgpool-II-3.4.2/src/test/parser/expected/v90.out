 SELECT "func"(a := 7,b := 12)
 SELECT "array_agg"("a" ORDER BY "b","c")
 SELECT "mleast"(VARIADIC ARRAY [10,-1,5,4.4])
 SELECT "avg"("c1") OVER ( ROWS 1 PRECEDING) FROM "r1"
 SELECT "avg"("c1") OVER "w1" FROM "r1" WINDOW "w1" AS ( ROWS BETWEEN 1 FOLLOWING AND 2 FOLLOWING)
 SELECT * FROM "t1" OFFSET 1 LIMIT 2
 SELECT * FROM "t1" LIMIT 2

pset 90000
EXPLAIN (analyze , verbose ) SELECT 1
EXPLAIN (analyze 'true', verbose 'true', costs 'true', buffers 'true', format 'xml') SELECT 1
EXPLAIN (analyze 'false', verbose 'false', costs 'false', buffers 'false', format 'json') SELECT 1
COPY "t" FROM STDIN 
COPY "t" FROM 'xxx' (format 'binary')
COPY "t" FROM 'xxx' (format 'csv', header 1, quote '#', escape '$', force_not_null ("foo", "bar"))
COPY "t" FROM 'xxx' (oids 1)
COPY "t" FROM 'xxx' (delimiter ',')
COPY "t" FROM 'xxx' (null 'x')
COPY "t" FROM STDIN (format 'text')
COPY "t" FROM STDIN (format 'csv')
COPY "t" FROM STDIN (format 'binary')
COPY "t" FROM 'f' (oids 'false')
COPY "t" FROM 'f' (oids 'true', delimiter '|', null '<N>', header 'false')
COPY "t" FROM 'f' (header 'true', quote '"', escape '@')
COPY "t" FROM 'f' (force_quote *, force_not_null ("one", "two", "three"))
